tests/publiccloud/ipa.pm

```diff
@@ -0,0 +1,162 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Use IPA framework to test public cloud SUSE images
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+use base "consoletest";
+use strict;
+use testapi;
+use utils;
+use File::Basename;
+use serial_terminal 'select_virtio_console';
+
+
+sub find_secgroup {
+    my ($name) = @_;
+
+    my $out = script_output("aws ec2 describe-security-groups --group-names '$name'", 30, proceed_on_failure => 1);
+    if ($out =~ /"GroupId":\s+"([^"]+)"/) {
+        return $1;
+    }
+    return;
+}
+
+sub save_logs {
+    my $output = script_output("find ipa_results -type f");
+    for my $file (split(/\n/, $output)) {
+        if ($file =~ m'ipa_results/ec2/ami-[0-9a-z]+/i-[0-9a-z]+/[0-9]{14}\.(log|results)') {
+            upload_logs($file);
```

Comment:
I think it makes sense to add failok=>1 , so you will get at least some logs in case of troubles

Reply:
ok will add this